source/codegen/metadata/nifgen/functions.py

```diff
@@ -0,0 +1,2994 @@
+            },
+            {
+                'name':'fileName',
+                'direction':'in',
+                'type':'ViConstString'
+            },
+            {
+                'name':'byteOrder',
+                'direction':'in',
+                'type':'ViInt32',
+                'enum':'ByteOrder'
+            },
+            {
+                'name':'waveformHandle',
+                'direction':'out',
+                'type':'ViInt32'
+            }
+        ],
+        'returns':'ViStatus'
+    },
+    'CreateWaveformFromFileHws':{
+        'cname' : 'niFgen_CreateWaveformFromFileHWS',
+        'parameters':[
+            {
+                'name':'vi',
+                'direction':'in',
+                'type':'ViSession'
+            },
+            {
+                'name':'channelName',
+                'direction':'in',
+                'type':'ViConstString'
+            },
+            {
+                'name':'fileName',
+                'direction':'in',
+                'type':'ViConstString'
+            },
+            {
+                'name':'useRateFromWaveform',
+                'direction':'in',
+                'type':'ViBoolean'
+            },
+            {
+                'name':'useGainAndOffsetFromWaveform',
+                'direction':'in',
+                'type':'ViBoolean'
+            },
+            {
+                'name':'waveformHandle',
+                'direction':'out',
+                'type':'ViInt32'
+            }
+        ],
+        'returns':'ViStatus'
+    },
+    'CreateWaveformFromFileI16':{
+        'parameters':[
+            {
+                'name':'vi',
+                'direction':'in',
+                'type':'ViSession'
+            },
+            {
+                'name':'channelName',
+                'direction':'in',
+                'type':'ViConstString'
+            },
+            {
+                'name':'fileName',
+                'direction':'in',
+                'type':'ViConstString'
+            },
+            {
+                'name':'byteOrder',
+                'direction':'in',
+                'type':'ViInt32',
+                'enum':'ByteOrder'
+            },
+            {
+                'name':'waveformHandle',
+                'direction':'out',
+                'type':'ViInt32'
+            }
+        ],
+        'returns':'ViStatus'
+    },
+    'CreateWaveformI16':{
+        'parameters':[
+            {
+                'name':'vi',
+                'direction':'in',
+                'type':'ViSession'
+            },
+            {
+                'name':'channelName',
+                'direction':'in',
+                'type':'ViConstString'
+            },
+            {
+                'name':'waveformSize',
+                'direction':'in',
+                'type':'ViInt32'
+            },
+            {
+                'name':'waveformDataArray',
+                'direction':'in',
+                'type':'ViInt16[]',
```

Comment:
I don't think we're supporting arrays of ViInt16 arrays right now in the generated code. So any method with a parameter of this type will generate a simple service handler to return `UNIMPLEMENTED` grpc status.

I think we haven't come accross the need to support this array type yet and there was some extra work to do in the generation to make sure the bits are carried to and from the API call and the client correctly.

If you add support for these types in a follow up PR, the first place to update is in [common_helpers.py is_unsupported_scalar_array](https://github.com/ni/grpc-device/blob/main/source/codegen/common_helpers.py#L37).

Reply:
Thanks. Removed 3 functions which were using the ViInt16 arrays.